### PR TITLE
Add RavenClientNoop, an empty (no-op) empty implementation of IRavenClient

### LINKED ISSUE #213: Noop implementation of IRavenClient
I think it would be nice to have a no-op implementation of IRavenClient available as that helps with dependency injection on development builds where I don't want to have nulls injected but rather an implementation that does not send data anywhere.

Would it make sense to include it in main distribution?

I can make a PR for this, however I am not sure about the naming of the class.

### PR DESCRIPTION
**Pull request by razzmatazz**
Fixes #213.

Note, that I am not sure about the value of RavenClientNoop.Dsn. Currently, I have it set to:

```c#
            currentDsn = new Dsn("http://localhost/no-op-dsn");
```

Which I am not sure is a right thing to do.

### CODE REVIEW

File: src/app/SharpRaven/RavenClientNoop.Net45.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+using SharpRaven.Data;
+
+namespace SharpRaven
+{
+    /// <summary>
+    /// Empty (no-op) implementation for the Raven Client for use in dependency injection
+    /// and other places when a silent operation is needed.
+    /// </summary>
+    public partial class RavenClientNoop
```
Comment (asbjornu) [inline src/app/SharpRaven/RavenClientNoop.Net45.cs]: Can we rename this to `NopRavenClient`, `NoOpRavenClient` or `NoopRavenClient`?

File: src/app/SharpRaven/RavenClientNoop.cs [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+            return Guid.NewGuid().ToString("n");
+        }
+
+
+        /// <summary>Sends the specified user feedback to Sentry</summary>
+        /// <returns>An empty string if succesful, otherwise the server response</returns>
+        /// <param name="feedback">The user feedback to send</param>
+        public string SendUserFeedback(SentryUserFeedback feedback)
+        {
+            return "";
```
Comment (asbjornu) [inline src/app/SharpRaven/RavenClientNoop.cs]: Can you change this to `string.Empty`?

**Comment by asbjornu** [comment]
I think the DSN is fine, but if you want, you can change it to the [reserved `invalid.` host name as per RFC 6761](https://tools.ietf.org/html/rfc6761#section-6.4)

**Comment by razzmatazz** [comment]
Thanks for the review.
I have now rebased the changes with class renamed to NoOpRavenClient.

**Comment by asbjornu** [comment]
Great, thanks! A pre-release containing this change should appear on NuGet.org in a few minutes.

**Comment by razzmatazz** [comment]
Great!

**Pull request merged**